### PR TITLE
Add setup docs for Zed editor

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

This PR adds the setup documentation for using Ruff with the Zed editor.

Closes: #12388

### CODE REVIEW

File: docs/editors/setup.md [hunk 1]
```diff
@@ -372,3 +372,125 @@ Alternatively, it can be used via the [Apheleia](https://github.com/radian-softw
...
+}
+```
+
+Taken together, you can configure Ruff to format, fix, and organize imports on-save via the
+following `settings.json`:
+
+!!! note
+    For this configuration, it is important to use the correct order of the formatter and code
+    action settings. The code actions should be defined before the formatter to ensure that the
+    formatter takes care of any remaining style issues after the code actions have been applied.
```
Comment (MichaReiser) [inline docs/editors/setup.md:473]: This note is unclear to me. The formatter isn't in code actions. How can the other code actions be before the formatter?

Comment (dhruvmanila) [reply docs/editors/setup.md:473]: What I meant was that in the `formatter` list, the code actions should be applied first, and then the language server should be used to run the formatter. So, the order in this list is important:
```
      "formatter": [
        {
          "code_actions": {
            "source.organizeImports.ruff": true,
            "source.fixAll.ruff": true
          }
        },
        {
          "language_server": {
            "name": "ruff"
          }
        }
      ]
```
Otherwise, if a list of unused imports are removed, the blank lines between the imports section and the following section remains.


https://github.com/user-attachments/assets/f3f00896-d249-4808-a79b-a776c9435621

Comment (MichaReiser) [reply docs/editors/setup.md:473]: ah, the language server sets the formatter. It's a bit confusing because the entire section is called *formatter*. Maybe rephrase to. ...the correct order of the code actions and the *formatter language server* settings.

File: docs/editors/setup.md [hunk 2]
```diff
@@ -372,3 +372,125 @@ Alternatively, it can be used via the [Apheleia](https://github.com/radian-softw
...
+        }
+      }
+    }
+  }
+}
+```
+
+!!! note
+    Configuring multiple formatters for a given language requires the [preview
+    version](https://zed.dev/releases/preview) of Zed.
```
Comment (osiewicz) [inline docs/editors/setup.md:422]: Maybe we should be specific about the version (0.146.0) multi-formatters were made available in instead? As is, it sounds like I have to use Preview to use multiple formatters, when in reality this change is gonna be available on Stable channel in a week.

Comment (dhruvmanila) [reply docs/editors/setup.md:422]: Yeah, that makes sense, thanks!

**Review by MichaReiser — APPROVED** [review]

**Comment by samueller** [comment]
I've read the setup docs, but still can't figure out how to add linting options. For example, where in Zed's `settings.json` do I add the option for [quote-style](https://docs.astral.sh/ruff/settings/#format_quote-style)? The docs give examples for `pyproject.toml` and `ruff.toml` and I've tried many possible places in `settings.json`, but can't figure it out.

**Comment by MichaReiser** [comment]
@samueller sorry that you're struggling with this. 

The editor doesn't support all ruff settings. It only supports a [subset](https://docs.astral.sh/ruff/editors/settings/) and `quote-style` isn't part of that subset. To configure the `quote-style`, you have to create a `pyproject.toml` or `ruff.toml` and set the quote style in there.

**Comment by samueller** [comment]
> It only supports a [subset](https://docs.astral.sh/ruff/editors/settings/) and `quote-style` isn't part of that subset.

I understand, thank you for letting me know

**Comment by dhruvmanila** [comment]
@samueller Sorry that you struggled with this. Do you have any recommendation on how can we improve this in the documentation to make it clearer?

**Comment by samueller** [comment]
> @samueller Do you have any recommendation on how can we improve this in the documentation to make it clearer?

Would have been great to have read which subset of features work in Zed's `settings.json` and which don't in the Zed configuration section.

**Comment by dhruvmanila** [comment]
> Would have been great to have read which subset of features work in Zed's `settings.json` and which don't in the Zed configuration section.

I'm curious to know whether the following section in https://docs.astral.sh/ruff/editors/setup/#zed conveys this:

> To configure the language server, you can provide the [server settings](https://docs.astral.sh/ruff/editors/settings/) under the [lsp.ruff.initialization_options.settings](https://zed.dev/docs/configuring-zed#lsp) key:
> ```json
> {
>   "lsp": {
>     "ruff": {
>       "initialization_options": {
>         "settings": {
>           // Ruff server settings goes here
>           "lineLength": 80,
>           "lint": {
>             "extendSelect": ["I"],
>           }
>         }
>       }
>     }
>   }
> }
> ```

**Comment by samueller** [comment]
> I'm curious to know whether the following section in https://docs.astral.sh/ruff/editors/setup/#zed conveys this:

I had seen that, but it didn't tell me the settings that won't work, such as `quote-reply`. I had assumed all settings would work, I just needed to figure out how to specify them in `settings.json`.
